lib/src/test/java/io/ably/lib/test/realtime/RealtimeJWTTest.java

```diff
@@ -0,0 +1,285 @@
+package io.ably.lib.test.realtime;
+
+import static org.junit.Assert.*;
+
+import io.ably.lib.debug.DebugOptions;
+import io.ably.lib.debug.DebugOptions.RawProtocolListener;
+import io.ably.lib.test.common.Setup.Key;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.ably.lib.types.*;
+import io.ably.lib.realtime.*;
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.rest.Auth.*;
+import io.ably.lib.test.common.Helpers.*;
+import io.ably.lib.test.common.ParameterizedTest;
+
+import java.util.UUID;
+
+
+public class RealtimeJWTTest extends ParameterizedTest {
+
+	private AblyRest restJWTRequester;
+	private ClientOptions jwtRequesterOptions;
+	private ClientOptions realtimeOptions;
+	private Key key = testVars.keys[0];
+	private final String clientId = "testJWTClientID";
+	private final String channelName = "testJWTChannel" + UUID.randomUUID().toString();
+	private final String messageName = "testJWTMessage" + UUID.randomUUID().toString();
+	Param[] keys = new Param[]{ new Param("keyName", key.keyName), new Param("keySecret", key.keySecret) };
+	Param[] clientIdParam = new Param[] { new Param("clientId", clientId) };
+	Param[] shortTokenTtl = new Param[] { new Param("expiresIn", 5) };
+	Param[] mediumTokenTtl = new Param[] { new Param("expiresIn", 35) };
+	private final String susbcribeOnlyCapability = "{\"" + channelName + "\": [\"subscribe\"]}";
+	private final String publishCapability = "{\"" + channelName + "\": [\"publish\"]}";
+
+	@Before
+	public void setUpBefore() throws Exception {
+		jwtRequesterOptions = createOptions(testVars.keys[0].keyStr);
+		jwtRequesterOptions.authUrl = echoServer;
```

Comment:
I don't like the fact that the same options instance is being used for every test. If necessary add a helper method to create an options and populate it, so you can easily create a new one each time.